src/Model/Sluggable/SluggableMethods.php

```diff
@@ -74,8 +74,17 @@ public function generateSlug()
             $usableValues = [];
 
             foreach ($fields as $field) {
-                // Too bad empty is a language construct...otherwise we could use the return value in a write context :)
-                $val = $this->{$field};
+                if (property_exists($this, $field)) {
+                    $val = $this->{$field};
+                } else {
+                    $methodName = 'get' . ucfirst($field);
+                    if (method_exists($this, $methodName)) {
+                        $val = $this->{$methodName}();
+                    } else {
+                        $val = null;
+                    }
+                }
```

Comment:
Maybe something like this would be more clear : 

```
$methodName = 'get'.ucfirst($field);

if (method_exists($this, $methodName)) {
    $val = $this->{$methodName}();
} elseif (property_exists($this, $field)) {
    $val = $this->{$field};
} else {
    continue;
}
```